### PR TITLE
New package for HPE-MPI using relocatable rpms

### PR DESCRIPTION
**Pull request by ohm314**
- Unpacks the archive and copies all files into prefix
- patches mpi compiler wrapper scripts to use `-isystem` rather than `-I` for include paths.

### CODE REVIEW

**Comment by matz-e** [comment]
Being tested right now (and for the next x hours)

**Review by matz-e — APPROVED** [review]
I've done minor modifications in our re-deployment to fit the new internals of spack, but this can be merged into our current branch!

File: var/spack/repos/builtin/packages/hpe-mpi/package.py [hunk 1]
```diff
@@ -6,12 +33,35 @@ class HpeMpi(Package):
     homepage = "http://www.example.com"
     url      = "http://www.example.com/hpempi-1.0.tar.gz"
 
-    version('2.16', '0123456789abcdef0123456789abcdef')
+    version('2.21', '2dd6c53a82993c4df929fdf898ac3d19',
+        url='file:///gpfs/bbp.cscs.ch/apps/hpc/download/hpe-mpi/hpe-mpi-2.21.tar.xz')
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/hpe-mpi/package.py:37]: can you sync with Dries to see how these can be uploaded to pulp server https://bbpteam.epfl.ch/repository/pulp/ ?

File: var/spack/repos/builtin/packages/hpe-mpi/package.py [hunk 2]
```diff
@@ -6,12 +33,35 @@ class HpeMpi(Package):
...
 
-    def do_fetch(self, mirror_only=False):
-        raise RuntimeError('HPE MPI should be configured as external package')
+
+    @run_before('install')
+    def unpack(self):
+        rpm2cpio = spack.util.executable.which('rpm2cpio')
+        cpio = spack.util.executable.which('cpio')
+
+        print(self.stage)
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/hpe-mpi/package.py]: to remove

File: var/spack/repos/builtin/packages/hpe-mpi/package.py [hunk 3]
```diff
@@ -6,12 +33,35 @@ class HpeMpi(Package):
...
+        for rpm_filename in find(self.stage.source_path, '*.rpm'):
+            print(rpm_filename)
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/hpe-mpi/package.py]: to remove

**Review by pramodk — CHANGES_REQUESTED** [review]
The MPI wrappers looks like below:

```
$ mpicxx -show
g++ -I/opt/hpe/hpc/mpt/mpt-2.16/include -lpthread /usr/lib64/libcpuset.so.1 /usr/lib64/libbitmask.so.1 -L/opt/hpe/hpc/mpt/mpt-2.16/lib -lmpi++ -lmpi
kumbhar@bbpv2:~$ mpicc -show
gcc -I/opt/hpe/hpc/mpt/mpt-2.16/include -lpthread /usr/lib64/libcpuset.so.1 /usr/lib64/libbitmask.so.1 -L/opt/hpe/hpc/mpt/mpt-2.16/lib -lmpi
```

can you update the compiler wrappers so that lib directory is added automatically to rpath? (so that compiled applications can run without loading hpe-mpi module).

File: var/spack/repos/builtin/packages/hpe-mpi/package.py (hunk 1 above)
Comment (ohm314) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:37]: OK, I'll check.

File: var/spack/repos/builtin/packages/hpe-mpi/package.py (hunk 2 above)
Comment (ohm314) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py]: You mean the print, right? indeed that got forgotten...

File: var/spack/repos/builtin/packages/hpe-mpi/package.py (hunk 1 above)
Comment (ohm314) [reply var/spack/repos/builtin/packages/hpe-mpi/package.py:37]: Let's stick with this URI for now

**Comment by ohm314** [comment]
retest please

**Comment by ohm314** [comment]
please retest

**Review by pramodk — APPROVED** [review]

**Pull request merged**